Sources/ServiceLifecycle/GracefulShutdown.swift

```diff
@@ -55,6 +55,18 @@ public func withGracefulShutdownHandler<T>(
     return try await operation()
 }
 
+/// Suspend the current task until graceful shutdown is executed.
+///
+/// A common use-case is to call this method in the `operation` closure of ``withGracefulShutdownHandler(operation:onGracefulShutdown:)``
+/// when you are only interested in executing a closure upon graceful shutdown, and nothing else.
+public func waitForGracefulShutdown() async {
```

Comment:
I think we should name this just `gracefulShutdown()` because you already call it with `await gracefulShutdown()`. This would be similar to what we have in XCTest here: https://developer.apple.com/documentation/xctest/xctestcase/4109476-fulfillment